packages/p2p-v2/src/hooks/__tests__/useQueryString.spec.tsx

```diff
@@ -0,0 +1,226 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import { Router, Route, Switch } from 'react-router-dom';
+import useQueryString from '../useQueryString';
+import { createMemoryHistory } from 'history';
+
+const MockBuySell = () => <h1>Buy sell page</h1>;
+const MockMyProfile = () => {
+    const { deleteQueryString, queryString, replaceQueryString, setQueryString } = useQueryString();
+
+    return (
+        <div>
+            <button
+                data-testid='switch_payment_methods_btn'
+                onClick={() =>
+                    setQueryString({
+                        tab: 'Payment methods',
+                    })
+                }
+            >
+                Switch payment methods
+            </button>
+            <button
+                data-testid='switch_stats_btn'
+                onClick={() =>
+                    setQueryString({
+                        tab: 'Stats',
+                    })
+                }
+            >
+                Switch stats
+            </button>
+            <button
+                data-testid='pm_form_btn'
+                onClick={() =>
+                    setQueryString({
+                        form: 'Payment method form',
+                        tab: 'Payment methods',
+                    })
+                }
+            >
+                Switch payment method form
+            </button>
+            <button
+                data-testid='counterparties_btn'
+                onClick={() =>
+                    replaceQueryString({
+                        form: 'Payment method form',
+                        tab: 'Counterparties',
+                    })
+                }
+            >
+                Switch counterparties
+            </button>
+            <button data-testid='go_back_btn' onClick={() => deleteQueryString('tab')}>
```

Comment:
for each of these buttons, i don't think we need to pass any test id's to it, we can just use getByRole instead right?

```suggestion
            <button onClick={() => deleteQueryString('tab')}>
```